Improvements

Keyrxng opened the pull request:
Resolves #19

Changes:
- readme display
- production config only
- reset to defaults

QA:



https://github.com/user-attachments/assets/20fb3e2c-994f-46be-8a1c-bb8e27e2678a

Keyrxng committed:
chore: prod only config

Keyrxng committed:
chore: fetch and render readme

Keyrxng committed:
feat: new dropdown, installed indicator

Keyrxng committed:
chore: load installed config values if present

Keyrxng committed:
feat: reset to defaults

Keyrxng committed:
chore: oauth redirect

Keyrxng committed:
chore: remove search parsing

Keyrxng committed:
chore: eslint and cleanup

Keyrxng committed:
chore: split renderer into files

Keyrxng committed:
chore: minor fixes

Keyrxng committed:
chore: markdown-it readme rendering, minor fixes

Keyrxng committed:
chore: feedback and await early

Keyrxng committed:
Merge remote-tracking branch 'upstream/main' into improvements

Keyrxng commented:
This PR became quite long so I will address tests in another task

@rndquu @gentlementlegen if you'd like to take a look when you can, cheers.

gentlementlegen commented:
Seems to work fine although I have a few remarks:
- I don't seem to have padding like in your video
<img width="1710" alt="image" src="https://github.com/user-attachments/assets/8fc12966-f72d-4798-aed1-3fc3ea06c81d">

- the left panel shifts a bit when hovering the "use default" button
- I am only seeing two orgs but I am in more orgs that that, how are they chosen?
- going back from the edit plugin menu makes the bar take the full width
<img width="1710" alt="image" src="https://github.com/user-attachments/assets/66799da9-95f9-4da0-9b4c-5e5183218dde">
- the "remove" button should maybe be disabled on plugins that are not installed

Keyrxng commented:
> I am only seeing two orgs but I am in more orgs that that, how are they chosen?

https://docs.github.com/en/rest/orgs/orgs?apiVersion=2022-11-28#list-organizations-for-the-authenticated-user

```ts
public async getGitHubUserOrgs(): Promise<string[]> {
    const octokit = await this.getOctokit();
    const response = await octokit.rest.orgs.listForAuthenticatedUser();
    return response.data.map((org: { login: string }) => org.login);
  }
```

What orgs did you see?

I've considered other methods but the question is: What orgs do you want to see? 

- Orgs that you are an admin of or just owner of, collaborator of?
- Should the partner determine which roles can edit the config or do we enforce rules, if we enforce then what should they be?

I think a task should be created and it discussed there because it'll involve additional work and direction.

We sort of scrapped global config variables from `ubiquity-os-config.yml` but we could leverage that to enforce partner defined roles which can edit (at least via the UI). 

E.g: The org owner must use it after `kernel-setup` and we allow them to input roles that can edit and we write this to the config separate from `plugins:` and we read this on fetch and check against our authenticated user.

Keyrxng committed:
chore: responsive padding, resize after nav

Keyrxng committed:
chore: disable remove if not installed

Keyrxng commented:
> the left panel shifts a bit when hovering the "use default" button

I couldn't reproduce this pls lmk if it happens again and what you suggest as a fix/steps to repro, cheers

Keyrxng committed:
chore: knip ignore

zugdev commented:
We should only show "no orgs available" after we've looked at the request answer, it can be confusing in high latency environments. Show "loading", either as string or animation. Perhaps copy from [work.ubq.fi](https://github.com/ubiquity/work.ubq.fi/blob/5ad2731d99f3c03cdb52fe5f7122e412669ca0bd/static/style/style.css#L817C5-L819C41).

zugdev commented:
1. Though I've selected an org there is no visual indicator of what org I am at, this can be confusing.

2. README fails to be opened for me. Also, no `console.error` log to help debug this.

![image](https://github.com/user-attachments/assets/5d4bf53e-28f1-49c8-af5f-6bd083ddd1f9)

Keyrxng committed:
chore: loading title on fetch

Keyrxng committed:
chore: inject org into editor title

Keyrxng commented:
> README fails to be opened for me. Also, no console.error log to help debug this.

Any chance you wrote a log to find out as I cannot repro and I don't think it happened for mentlegen either or he'd have mentioned it?

I can add a console.error but would like to know the root cause as me adding a log won't resolve it, appreciate if you could.

> We should only show "no orgs available"

I thought this notification was enough but I have updated the title for while it's fetching also

![image](https://github.com/user-attachments/assets/c33cdd69-97e0-4500-84c3-7ece61e86495)

zugdev commented on static/main.ts:
```diff
@@ -12,30 +13,33 @@ async function handleAuth() {
+        renderOrgPicker(renderer, []);
+
         const manifestCache = await fetcher.fetchMarketplaceManifests();
         localStorage.setItem("manifestCache", JSON.stringify(manifestCache));
```
This is a duplicate cache set, you already cache manifests inside `fetchMarketplaceManifests()` on L41 of `fetch-manifest.ts`:

`localStorage.setItem("manifestCache", JSON.stringify(manifestCache));`

zugdev commented on static/scripts/rendering/plugin-select.ts:
```diff
@@ -0,0 +1,93 @@
+  renderer.manifestGuiBody.innerHTML = null;
+  controlButtons({ hide: true });
+
+  const manifestCache = JSON.parse(localStorage.getItem("manifestCache") || "{}") as ManifestCache;
```
You have `JSON.parse(localStorage.getItem("manifestCache") || "{}")` more than once, perhaps write a getter function.

zugdev commented:
These icons are not intuitive, perhaps text is better.

![image](https://github.com/user-attachments/assets/a6b2903d-c8d1-47a3-9c1f-7c3b45da1bba)

zugdev commented on static/index.html:
```diff
@@ -26,7 +26,7 @@
           <tfoot
             ><tr
-              ><td></td><td><button id="remove"></button><button id="add"></button></td></tr
+              ><td></td><td><button id="reset-to-default"></button><button id="remove"></button><button id="add"></button></td></tr
```
If there is still time we should move buttons outside of table. Tables are terrible for layouting and will definitely generate padding/overflow problems eventually. I think the config itself is fine within table as there is no easy better way to do it.

Keyrxng replied:
Move them to where? I lack the imagination lmao. Perhaps the little sprinkles and sparkles can be handled as one PR once this is stable? https://github.com/ubiquity-os/ubiquity-os-plugin-installer/issues/14 will be the last change to the UI for for V1

Keyrxng commented:
> These icons are not intuitive, perhaps text is better.
> 
> ![image](https://private-user-images.githubusercontent.com/155616000/389068174-a6b2903d-c8d1-47a3-9c1f-7c3b45da1bba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIzMDI4ODQsIm5iZiI6MTczMjMwMjU4NCwicGF0aCI6Ii8xNTU2MTYwMDAvMzg5MDY4MTc0LWE2YjI5MDNkLWM4ZDEtNDdhMy05YzFmLTdjM2I0NWRhMWJiYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyMlQxOTA5NDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNjRlMjM0MzY5NjAyMmJmZWIyMjkwYmExYmQzNTZkY2NlNDJkZDk3YjE0NTRkNDI4ZDIxYTNlZjk5ZDUwOGYyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.UL5hYYRPZZV1P2MckH1ueBPUgH1FvEb0taMu_CehkPk)

They do have text when hovered and were part of 0x4007's original implementation I just tried to adopt a similar approach for reset to defaults but if you think just text would be better I can do that lmk.

![image](https://github.com/user-attachments/assets/58902fac-f05b-4481-bc73-404fffe8de59)

zugdev commented:
Some responsiveness issues, not sure if relevant as of this PR:

![image](https://github.com/user-attachments/assets/5dc23202-332c-4f0d-b932-8f76e1c25919)

zugdev commented:
json is not vertically expanding for me:

https://github.com/user-attachments/assets/cc0b8c1c-e623-47bf-a8cf-0e0822d7197e

Keyrxng committed:
chore: storage getters

zugdev commented:
Once a plugin config is added, a popup appears to push to GitHub. I think we either directly push (kinda dangerous), or offer a "shopping cart" where changes get added to your "cart" and partner reviews it all before pushing.

![image](https://github.com/user-attachments/assets/771a97f4-12b3-4a81-8415-59d325f11007)

Also, if I push and go back the plugin won't appear as green/added until I refresh:

https://github.com/user-attachments/assets/3405fd4f-b263-4b24-8e39-8f86d29fccb5

Keyrxng commented:
> json is not vertically expanding for me:
> 
>  d.mp4

1. That is relevant to https://github.com/ubiquity-os/ubiquity-os-plugin-installer/issues/14. If you don't have that plugin installed all of those fields render as `null` and specific handling has not been implemented yet.
2. You'll notice that it does expand on other plugins

> Once a plugin config is added, a popup appears to push to GitHub. I think we either directly push (kinda dangerous), or offer a "shopping cart" where changes get added to your "cart" and partner reviews it all before pushing.

If you ignore or close the notification you should be able to continue building other plugins continuously, so there is a shopping cart in effect but hidden. Think it's worth a little legend display what plugins are stored but not yet pushed?

> Some responsiveness issues, not sure if relevant as of this PR:

Not sure exactly what you are referring to, I assume the height mismatch? 

Some plugins render a < 250px modal, `convo-rewards` is the largest and is going to become much larger after https://github.com/ubiquity-os/ubiquity-os-plugin-installer/issues/14, so just introduce a basic scroll for the editor and capped it's size to the readme?

zugdev commented:
> Any chance you wrote a log to find out as I cannot repro and I don't think it happened for mentlegen either or he'd have mentioned it?

Cleared cache and is now working smooth, perhaps it was on my end. If I can repro again I'll debug.

> I thought this notification was enough but I have updated the title for while it's fetching also

I think styling decisions are more opinion based comments than direct request for changes. Just let it be for now, unless agreed by team.

Keyrxng committed:
chore: indicator after push without reload

Keyrxng committed:
chore: format

Keyrxng commented:
> Also, if I push and go back the plugin won't appear as green/added until I refresh:

Resolved here [7bd56ac](https://github.com/ubiquity-os/ubiquity-os-plugin-installer/pull/25/commits/7bd56acec90ae8d1a3bf12f608094dc778ed08f9)

> > I thought this notification was enough but I have updated the title for while it's fetching also

> I think styling decisions are more opinion based comments than direct request for changes. Just let it be for now, unless agreed by team.

I implemented the title change alongside the notification but I'm going to leave the buttons alone unless told that it should just be text. 

As for button placement, I think they look good where they are but if I'm told to place them somewhere specific I can do that.

Keyrxng commented:
> @Keyrxng, this task has been idle for a while. Please provide an update.

waiting for the team

gentlementlegen commented:
> > I am only seeing two orgs but I am in more orgs that that, how are they chosen?
> 
> https://docs.github.com/en/rest/orgs/orgs?apiVersion=2022-11-28#list-organizations-for-the-authenticated-user
> 
> ```ts
> public async getGitHubUserOrgs(): Promise<string[]> {
>     const octokit = await this.getOctokit();
>     const response = await octokit.rest.orgs.listForAuthenticatedUser();
>     return response.data.map((org: { login: string }) => org.login);
>   }
> ```
> 
> What orgs did you see?
> 
> I've considered other methods but the question is: What orgs do you want to see?
> 
> * Orgs that you are an admin of or just owner of, collaborator of?
> * Should the partner determine which roles can edit the config or do we enforce rules, if we enforce then what should they be?
> 
> I think a task should be created and it discussed there because it'll involve additional work and direction.
> 
> We sort of scrapped global config variables from `ubiquity-os-config.yml` but we could leverage that to enforce partner defined roles which can edit (at least via the UI).
> 
> E.g: The org owner must use it after `kernel-setup` and we allow them to input roles that can edit and we write this to the config separate from `plugins:` and we read this on fetch and check against our authenticated user.

I have seen only 2 out of 4 orgs where I am an admin with the rights to create a configuration (`Meniole` and `ubiquity-os`). I should have seen `ubiquity-os-marketplace` most likely.

Keyrxng commented:
> I have seen only 2 out of 4 orgs where I am an admin with the rights to create a configuration (Meniole and ubiquity-os). I should have seen ubiquity-os-marketplace most likely.

![image](https://github.com/user-attachments/assets/ade27502-3b4b-41fb-9d17-f195ec8d99ad)

![image](https://github.com/user-attachments/assets/6b4f50bd-a85e-4daa-bf88-a488c54e95b2)

You are an `Owner` in both of those orgs @gentlementlegen so I'd expect marketplace to show up too but idk why it isn't. I don't share your permissions so I cannot debug this personally because any other org I use I'll be the creator of lmao.

What are your thoughts? 

- Marketplace org permission/access settings affecting the API res?
- wrong API method being used?

---

I did try `listPermissionsforUser` which lists `member`-`admin` etc but the gotcha is that an owner shares the same role as an admin so it's not straight forward to determine if they own it.

Keyrxng commented:
>  I should have seen ubiquity-os-marketplace most likely.

So this UI should display all orgs that the user is an admin of then? Or simply all orgs in which you have `write` permissions?

If I know clearly what orgs should be displayed (defined by role or some other tangible fact) then I can make efforts to make it happen.

zugdev commented:
Of course the ideal is to show precisely the ones he can edit. If that can't be done it's better to show all admin orgs than to miss some.

zugdev commented on static/scripts/rendering/org-select.ts:
```diff
@@ -0,0 +1,99 @@
+    optionDiv.addEventListener("click", () => {
+      handleOrgSelection(renderer, org);
+      selectSelected.textContent = org;
+      localStorage.setItem("selectedOrg", org);
```
redundant cache set here, handleOrgSelection already caches

zugdev commented:
I can't add conversation rewards plugin with default config, I imagine it's because of the undefineds and nulls. An error message should indicate that.

[cant-add-conversation-rewards.webm](https://github.com/user-attachments/assets/ef7627af-062d-4f06-a0bf-cdbdd6e48d2a)

zugdev commented:
When trying to add activity watcher, 5 errors pop up at the same time regarding the plugin URL being unavailable.

[watcher.webm](https://github.com/user-attachments/assets/b1adc185-6c62-4254-9758-75ad01867e30)

zugdev commented:
When trying to add wallet, 7 success pop ups. I am thinking the pop ups are accumulating, will investigate.

![image](https://github.com/user-attachments/assets/f0f24024-a3b5-40cd-aca6-567014b6f41f)

zugdev commented:
writeNewConfig executions are accumulating:

[accumulating-toaster.webm](https://github.com/user-attachments/assets/de6d1c8b-f95d-490d-bdf1-8c88254ae800)

It's outside of toaster, you can follow the experiment by adding `console.log("1");` inside of `writeNewConfig(...)` function declaration. Then go into a plugin add it, go back, add it go back, it follows an arithmetic progression each time you do that.

[accumulating-writeConfigs.webm](https://github.com/user-attachments/assets/4e449e78-8fb4-44ab-bf18-bb522a861e60)

zugdev commented on static/scripts/rendering/config-editor.ts:
```diff
@@ -0,0 +1,109 @@
+  if (!add || !remove) {
+    throw new Error("Add or remove button not found");
+  }
+  add.addEventListener("click", writeNewConfig.bind(null, renderer, "add"));
```
the problem is this addEventListener statement, if only you navigate a couple times and then execute an action it will stackup. you only need one event listener please check that. 

[eventListenerAdded.webm](https://github.com/user-attachments/assets/976db9d7-b09b-42c5-9663-72d10b858773)

zugdev commented:
Remove is broken for me, nothing happens when I press it.

zugdev commented on static/scripts/rendering/write-add-remove.ts:
```diff
@@ -0,0 +1,132 @@
+
+function handleAddPlugin(renderer: ManifestRenderer, plugin: Plugin, pluginManifest: Manifest): void {
+  renderer.configParser.addPlugin(plugin);
+  toastNotification(`Configuration for ${pluginManifest.name} saved successfully.Do you want to push to GitHub ? `, {
```
typo: space between sentences here

zugdev commented on static/scripts/rendering/write-add-remove.ts:
```diff
@@ -0,0 +1,132 @@
+
+  if (option === "add") {
+    handleAddPlugin(renderer, plugin, pluginManifest);
+  } else {
```
please use `else if (option === "remove")`, if new option is added someday can lead to error

zugdev commented on static/scripts/rendering/write-add-remove.ts:
```diff
@@ -0,0 +1,132 @@
+
+function handleRemovePlugin(renderer: ManifestRenderer, plugin: Plugin, pluginManifest: Manifest): void {
+  renderer.configParser.removePlugin(plugin);
+  toastNotification(`Configuration for ${pluginManifest.name} removed successfully.Do you want to push to GitHub ? `, {
```
typo: space between sentences

zugdev commented on static/scripts/rendering/config-editor.ts:
```diff
@@ -0,0 +1,109 @@
+  if (plugin) {
+    remove.disabled = false;
+    remove.classList.remove("disabled");
+    remove.addEventListener("click", () => writeNewConfig.bind(null, renderer, "remove"));
```
the function isn't being called because it's in a arrow function. please copy the syntax from the add button:

 remove.addEventListener("click", writeNewConfig.bind(null, renderer, "remove"));

zugdev commented on static/scripts/rendering/config-editor.ts:
```diff
@@ -0,0 +1,109 @@
+    throw new Error("Reset to default button not found");
+  }
+
+  resetToDefaultButton.addEventListener("click", () => {
```
pressing reset to default is disabling the remove button when it shouldn't please mind that you should pass in plugin to `renderConfigEditor`

Keyrxng committed:
chore: add/remove shift

Keyrxng committed:
chore: editor listeners, active remove for installed

Keyrxng committed:
chore: process typebox unions for convo-rewards

Keyrxng committed:
chore: remove dupe storage set

Keyrxng committed:
chore: toast typos

Keyrxng committed:
chore: tracked listeners, normalize util

rndquu reviewed (changes requested):
@Keyrxng 

1. Getting this error on trying to add the https://github.com/ubiquity-os-marketplace/text-conversation-rewards plugin:
<img width="1511" alt="Screenshot 2024-11-26 at 11 50 32" src="https://github.com/user-attachments/assets/242b2bc1-91ef-4ec3-9fd5-53a739ec8adb">
2. Getting this error o trying to add the https://github.com/ubiquity-os-marketplace/daemon-pricing plugin:
<img width="1511" alt="Screenshot 2024-11-26 at 11 51 11" src="https://github.com/user-attachments/assets/bae62f13-a936-45eb-b304-24e3da1eabeb">

Pls fix

Keyrxng committed:
chore: required fields input assert and display

Keyrxng committed:
chore: more robust readme fetching, cleanup, input type assertion

Keyrxng committed:
chore: comments

Keyrxng commented:
Both of those plugins fixed: `pricing` was improperly setting of null/undefined vals and `conversation-rewards` was due to the handling of the huge config.

I've just JSON.stringified the huge props for now but I think you are right in that there is no clear clean way of displaying it.

It's messy when you are having to work with `undefined` and `null` values because they should represent a clear fact like `null === they need it but don't have it` and `undefined === can be omitted` but they are used interchangeably

Manually expanded:
![image](https://github.com/user-attachments/assets/a48bed13-f6a8-491e-b0ac-17b865bee11d)

QA of both plugins:
https://github.com/user-attachments/assets/373142d5-76db-4556-81cd-323df09faf5c

Keyrxng commented:
> Of course the ideal is to show precisely the ones he can edit. If that can't be done it's better to show all admin orgs than to miss some.

`rest.orgs.listForUser():`
> List [public organization memberships](vscode-file://vscode-app/c:/Users/keyrx/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) for the specified user.
> This method only lists public memberships, regardless of authentication. If you need to fetch all of the organization memberships (public and private) for the authenticated user, use the [List organizations for the authenticated user](vscode-file://vscode-app/c:/Users/keyrx/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) API instead.

This returns all 4 orgs for me and @gentlementlegen, but returns 0 for @zugdev due to your org membership not being public. So we cannot rely on the public `listForUser` endpoints if it does not capture private memberships.

`rest.orgs.listMembershipsForAuthenticatedUser()`:
I set full access with: `scopes: "user repo org"` and it _still_ only returns my org `ubq-testing`. I do have write permissions in `ubiquity` still so it should be appearing for me but isn't.

I think org display should be handled in #26 rather than hold this back but appreciate the clarification for you both, should display all orgs with `write` permissions.

Keyrxng commented:
This error used to occur on `work.ubq.fi` to me often although it doesn't happen on there anymore, I'm not sure if it was resolved specifically or if it's because of something else, @zugdev do you have any knowledge of this happening on `work.ubq` or was this before your time?

![image](https://github.com/user-attachments/assets/266ff356-61e6-4640-a9ba-2acfbd3b95a7)


---

This is the code that's throwing for me. I assume it's throwing because I'm being specific in the method and when the list methods are used those that would throw are just omitted.

```ts
  public async getGitHubUserOrgs(): Promise<{ org: string; permissions: Record<string, unknown> | undefined, role: string }[]> {
    const octokit = await this.getOctokit();
    const response = await octokit.rest.orgs.listForAuthenticatedUser();
    const authedUser = await octokit.rest.users.getAuthenticated();
    const orgPermission = await octokit.rest.orgs.listMembershipsForAuthenticatedUser();
    const orgs = await octokit.rest.orgs.listForUser({ username: authedUser.data.login });

    const userOrgs = orgPermission.data.map((org) => {
      return {
        org: org.organization.login,
        permissions: org.permissions,
        role: org.role,
      };
    })

    const publicOrgs = []

    for (const org of orgs.data) {
      const orgUserDetails = await octokit.rest.orgs.getMembershipForAuthenticatedUser({
        org: org.login,
      });

      publicOrgs.push({
        org: org.login,
        permissions: undefined,
        role: undefined,
      })
    }

    console.log("User orgs", userOrgs);

    return response.data;
  }
```

Keyrxng commented:
Considering the above two comments, I'm re-opening this for review and the org selection discussion can be moved into #26

zugdev commented:
> This error used to occur on `work.ubq.fi` to me often although it doesn't happen on there anymore, I'm not sure if it was resolved specifically or if it's because of something else, @zugdev do you have any knowledge of this happening on `work.ubq` or was this before your time?
> 
> ![image](https://private-user-images.githubusercontent.com/106303466/389922097-266ff356-61e6-4640-a9ba-2acfbd3b95a7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI2MzY5NDAsIm5iZiI6MTczMjYzNjY0MCwicGF0aCI6Ii8xMDYzMDM0NjYvMzg5OTIyMDk3LTI2NmZmMzU2LTYxZTYtNDY0MC1hOWJhLTJhY2ZiZDNiOTVhNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyNlQxNTU3MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NmI3M2EzZTI1NDUyOTZmZWI5MzM2M2RhY2I1NmYwMjUyYmJkMDU5MjZkMDE3ZTdlMTI3OWY0ZWJlMzRhNWM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.7Ni1493sNIfmvhKlYwFLmZYRYYCVa_cinSWPTCP79zA)

This is oAuth rate limit.

zugdev commented:
Appears like you have solved the problems I found. The only thing I managed to find now is the overlap between table and header:

![image](https://github.com/user-attachments/assets/a02b0e61-fddd-4c79-b3e7-4078602c3d1e)

We could create a issue later for me to fix these kind of CSS problems. By taking a very quick look:

1. Nav should be inside of header
2. The height for content does not need to be calculated just use display block use the available height
3. Fix overlapping
4. Make responsive

Keyrxng commented:
> We could create a issue later for me to fix these kind of CSS problems. By taking a very quick look:

I'll catch what I can during reviews in the remaining three PRs but yeah that would be good mate cheers.

rndquu reviewed (changes requested):
@Keyrxng Check this screenshot:
<img width="1511" alt="Screenshot 2024-11-27 at 18 46 00" src="https://github.com/user-attachments/assets/50013049-ba69-4659-83df-36da403f630c">

The "Assistive pricing" plugin is disabled although its config is present in the `.ubiquity-os.config.yml` file. The expected behavior for the "Assistive pricing" is to be in the enabled mode.

Pls fix

Keyrxng committed:
chore: fix config save

Keyrxng commented:
> @Keyrxng Check this screenshot: 
> 
> The "Assistive pricing" plugin is disabled although its config is present in the `.ubiquity-os.config.yml` file. The expected behavior for the "Assistive pricing" is to be in the enabled mode.
> 
> Pls fix

My apologies @rndquu, fixed now

Keyrxng committed:
chore: auto dismiss add/remove plugin notification

Keyrxng committed:
chore: single push method, commit title update

rndquu reviewed (changes requested):
@Keyrxng 

1. For some of the plugins (`User activity watcher`, `Automated merging`, `Telegram kernel`) getting the `No plugin URL found for user-activity-watcher` error:
<img width="1511" alt="Screenshot 2024-11-28 at 10 24 57" src="https://github.com/user-attachments/assets/505e9476-8289-4b85-83a3-708627d09267">

As far as I understand those plugins work as github actions so it's not clear why we can't set them up.

2. The `command-ask` plugin shows empty UI:
<img width="1511" alt="Screenshot 2024-11-28 at 10 29 42" src="https://github.com/user-attachments/assets/f3cbec23-00ab-496d-8fc0-c62cd3497064">

Pls fix

Keyrxng commented:
Sorted @rndquu

1. PRs opened in relevant repos: related to #27
2. I'm not able to reproduce this. Was it from a fresh cache fetch or did you have it installed?
![image](https://github.com/user-attachments/assets/71ee4427-f6cf-4a5b-99f4-9ee2cd33d16f)

Keyrxng commented:
I merged https://github.com/ubiquity-os-marketplace/ubiquity-os-kernel-telegram/pull/40 and `kernel telegram` is now working so the other PRs need merged and we should be good to merge this?

![image](https://github.com/user-attachments/assets/4524aae3-c175-44c6-8edf-be156562fc4c)


1. Technically `kernel-telegram` is not built to support being installed by partners at the moment so should probs not be displayed.
2. It has two config entries: A) I detect url(s), if plural, create an entry for each B) See links below

- https://github.com/ubiquity-os/plugin-template/issues/33#issuecomment-2500425415
- https://github.com/ubiquity-os-marketplace/ubiquity-os-kernel-telegram/issues/20

rndquu reviewed (changes requested):
Check this screenshot:
<img width="1511" alt="Screenshot 2024-11-28 at 19 41 55" src="https://github.com/user-attachments/assets/80f20e40-8266-48fd-9860-afa85b3f2ca8">

Although the `registerWalletWithVerification` is set to `true` in the `yml` config the option in the html UI is not check while the expected behavior is to load current config values.

Pls fix.

rndquu commented:
> Sorted @rndquu
> 
> 1. PRs opened in relevant repos: related to [`manifest.name` mismatch #27](https://github.com/ubiquity-os/ubiquity-os-plugin-installer/issues/27)
> 2. I'm not able to reproduce this. Was it from a fresh cache fetch or did you have it installed?
>    ![image](https://private-user-images.githubusercontent.com/106303466/390768310-71ee4427-f6cf-4a5b-99f4-9ee2cd33d16f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI4MTI2NTAsIm5iZiI6MTczMjgxMjM1MCwicGF0aCI6Ii8xMDYzMDM0NjYvMzkwNzY4MzEwLTcxZWU0NDI3LWY2Y2YtNGE1Yi05OWY0LTllZTJjZDMzZDE2Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyOFQxNjQ1NTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMjk0Nzg0NDRlZTM2NWYyOTUxMmQ1YjlkYjhlNTc2MjQ3ZjQ4NGJmODMwNjcyYjhmZDMwMzI5ODJmMTNjODBkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.fXELk9RrKZ0PiVGgCDsKg1T3p4XMHWtyGOJdAdlcPDY)

1. So, as far as I understand, when https://github.com/ubiquity-os/ubiquity-os-plugin-installer/issues/24 is solved all plugins setup will be fine because of the correct `name` property.
2. I cleared the cache and error is gone

Keyrxng committed:
fix: check for boolean input on user config injection

Keyrxng commented:
Fixed, I had forgot to set `checked` relevant to the input value.

![image](https://github.com/user-attachments/assets/6732b0e4-2e5e-442c-9abd-ace87ab129c6)


> So, as far as I understand, when https://github.com/ubiquity-os/ubiquity-os-plugin-installer/issues/24 is solved all plugins setup will be fine because of the correct name property.
I cleared the cache and error is gone

Yeah as can be seen with `kernel-telegram` after I merged. Excellent re: cache issue, I tried to improve the fetching of the readme because some would `404` maybe the manifest fetching needs made more robust if we see anything like that again

rndquu commented:
> Appears like you have solved the problems I found. The only thing I managed to find now is the overlap between table and header:
> 
> ![image](https://private-user-images.githubusercontent.com/155616000/390039048-a02b0e61-fddd-4c79-b3e7-4078602c3d1e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI4MTM5NjUsIm5iZiI6MTczMjgxMzY2NSwicGF0aCI6Ii8xNTU2MTYwMDAvMzkwMDM5MDQ4LWEwMmIwZTYxLWZkZGQtNGM3OS1iM2U3LTQwNzg2MDJjM2QxZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyOFQxNzA3NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NDM5NDZlNmVhYWIyMmQ4YWFjMzQzYmMzZThjM2UzNWU0MmRiM2ExZTM5MzE3MTYyNWVhYWE1ZjM1Njg5M2I5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.4mtfTaoFTLP-uaixXhdsyzsvuNgggi6DXYTYgOEsaSU)
> 
> We could create a issue later for me to fix these kind of CSS problems. By taking a very quick look:
> 
> 1. Nav should be inside of header
> 2. The height for content does not need to be calculated just use display block use the available height
> 3. Fix overlapping
> 4. Make responsive

I suppose when all of the critical issues are ready (https://github.com/ubiquity-os/ubiquity-os-plugin-installer/issues/24, https://github.com/ubiquity-os/ubiquity-os-plugin-installer/issues/22, https://github.com/ubiquity-os/ubiquity-os-plugin-installer/issues/20) we can get back to neaten up the css.

rndquu commented:
@Keyrxng Now you could merge the latest `main` branch into your other PRs from https://github.com/ubiquity-os/ubiquity-os-plugin-installer/pulls
